sound/soc/sof/sof-pci-dev.c

```diff
@@ -227,6 +252,33 @@ static int sof_pci_probe(struct pci_dev *pci,
 	sof_pdata->dev = dev;
 	sof_pdata->platform = dev_name(dev);
 
+	/* alternate fw and tplg filenames ? */
+	if (fw_path)
+		path_prefix = fw_path;
+	else
+		path_prefix = sof_pdata->desc->default_fw_path;
+
+	fw_filename = devm_kasprintf(dev, GFP_KERNEL,
+				     "%s/%s",
+				     path_prefix,
+				     sof_pdata->machine->sof_fw_filename);
```

Comment:
@plbossart this will definitely break HDA platoforms as sof_pdata->machine will be NULL.

Reply:
ah yes, this entire part should be conditional the machine being detected.
That's really why I wanted the Up with HDMI only working so that we can test this path. I don't have access to any other platform.

Reply:
fixed and tested on Up2.